scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -0,0 +1,64 @@
+use crate::compiler::plugin::proc_macro::ProcMacroInstance;
+use crate::core::Package;
+use anyhow::Result;
+use cairo_lang_defs::plugin::{MacroPlugin, MacroPluginMetadata, PluginResult};
+use cairo_lang_semantic::plugin::PluginSuite;
+use cairo_lang_syntax::node::ast::ModuleItem;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use smol_str::SmolStr;
+use std::collections::HashMap;
+use std::sync::Arc;
+use typed_builder::TypedBuilder;
+
+/// A Cairo compiler plugin controlling the procedural macro execution.
+///
+/// This plugin decides which macro plugins (if any) should be applied to the processed AST item.
+/// It then redirects the item to the appropriate macro plugin for code expansion.
+#[derive(Debug, TypedBuilder)]
+pub struct ProcMacroHost {
+    macros: HashMap<SmolStr, Arc<ProcMacroInstance>>,
+}
+
+impl MacroPlugin for ProcMacroHost {
+    fn generate_code(
+        &self,
+        _db: &dyn SyntaxGroup,
+        _item_ast: ModuleItem,
+        _metadata: &MacroPluginMetadata<'_>,
+    ) -> PluginResult {
+        // Apply expansion to `item_ast` where needed.
+        // TODO(maciektr): Implement
+        PluginResult::default()
+    }
+
+    fn declared_attributes(&self) -> Vec<String> {
+        self.macros.keys().map(|name| name.to_string()).collect()
```

Comment:
A single instance will be capable of providing multiple macros, so I think you'd better be with having `declared_attributees` method in the instance; instead of binding it to name

Reply:
Yep, I just assumed we would change it later, as we have deprioritized the multiple macros per package feature :D  
Will add it now